server/models/Collective.js

```diff
@@ -1265,134 +1265,133 @@ export default function(Sequelize, DataTypes) {
    * @param {*} role
    * @param {*} defaultAttributes
    */
-  Collective.prototype.addUserWithRole = function(user, role, defaultAttributes = {}, transaction) {
+  Collective.prototype.addUserWithRole = async function(user, role, defaultAttributes = {}, context = {}, transaction) {
     if (role === roles.HOST) {
       return logger.info('Please use Collective.addHost(hostCollective, remoteUser);');
     }
 
     const sequelizeParams = transaction ? { transaction } : undefined;
 
-    const member = {
+    const memberAttributes = {
       role,
       CreatedByUserId: user.id,
       MemberCollectiveId: user.CollectiveId,
       CollectiveId: this.id,
       ...defaultAttributes,
     };
 
-    debug('addUserWithRole', user.id, role, 'member', member);
-    return Promise.all([
-      models.Member.create(member, sequelizeParams),
-      models.User.findByPk(
-        member.CreatedByUserId,
+    debug('addUserWithRole', user.id, role, 'member', memberAttributes);
+
+    const member = await models.Member.create(memberAttributes, sequelizeParams);
+
+    switch (role) {
+      case roles.BACKER:
+      case roles.ATTENDEE:
+        if (!context.skipActivity) {
+          await this.createCollectiveMemberCreatedActivity(member, context, sequelizeParams);
+        }
+        break;
+
+      case roles.MEMBER:
+      case roles.ADMIN:
+        await this.sendNewMemberEmail(user, role, member, sequelizeParams);
+        break;
+    }
+
+    return member;
+  };
+
+  Collective.prototype.createCollectiveMemberCreatedActivity = async function(member, context, sequelizeParams) {
```

Comment:
I find that name quite verbose, because we're in 'Collective' I feel like `createMemberCreatedActivity` would be enough

Reply:
I agree but I wanted to make it consistent. Will change it I think.